frontend/src/pages/modelRegistry/screens/ModelVersionDetails/ModelVersionRegisteredFromLink.tsx

```diff
@@ -0,0 +1,174 @@
+import PopoverListContent from '~/components/PopoverListContent';
+import { globalPipelineRunDetailsRoute, getCatalogModelDetailsRoute } from '~/routes';
+import { ModelArtifact } from '~/concepts/modelRegistry/types';
+import {
+  modelSourcePropertiesToCatalogParams,
+  modelSourcePropertiesToPipelineRunRef,
+} from '~/concepts/modelRegistry/utils';
+
+/**
+ * Props for the ModelVersionRegisteredFromLink component
+ */
+type ModelVersionRegisteredFromLinkProps = {
+  modelArtifact: ModelArtifact;
+};
+
+/**
+ * Component that displays where a model version was registered from
+ * Uses utility functions to determine if it was from a pipeline run or catalog model
+ */
+const ModelVersionRegisteredFromLink: React.FC<ModelVersionRegisteredFromLinkProps> = ({
+  modelArtifact,
+}) => {
+  const { projects } = React.useContext(ProjectsContext);
+
+  // Extract information using utility functions
+  const registeredFromCatalogDetails = modelSourcePropertiesToCatalogParams(
+    modelArtifact.modelSourceKind,
+    modelArtifact.modelSourceClass,
+    modelArtifact.modelSourceGroup,
+    modelArtifact.modelSourceName,
+    modelArtifact.modelSourceId,
+  );
+
+  const registeredFromPipelineDetails = modelSourcePropertiesToPipelineRunRef(
+    modelArtifact.modelSourceKind,
+    modelArtifact.modelSourceGroup,
+    modelArtifact.modelSourceId,
+    modelArtifact.modelSourceName,
+  );
+
+  // If no source information is available, don't render anything
+  if (!registeredFromCatalogDetails && !registeredFromPipelineDetails) {
+    return null;
+  }
+
+  // Generate the catalog URL if we have catalog details
+  const catalogModelUrl = registeredFromCatalogDetails
+    ? getCatalogModelDetailsRoute(registeredFromCatalogDetails)
+    : '';
+
+  // For pipeline source, look up the project
+  const registeredPipelineProject =
+    registeredFromPipelineDetails &&
+    projects.find((p) => p.metadata.name === registeredFromPipelineDetails.project);
+
+  // If we have pipeline details, render the pipeline link
+  if (registeredFromPipelineDetails) {
+    const { project, runId, runName } = registeredFromPipelineDetails;
+
+    // If we also have a catalog URL, link to the catalog
+    const renderRunLink = catalogModelUrl ? (
+      <Link
+        to={catalogModelUrl}
+        style={{ fontWeight: 'var(--pf-t--global--font--weight--body--bold)' }}
+        data-testid="catalog-model-link"
+      >
+        {runName}
+      </Link>
+    ) : registeredPipelineProject ? (
+      <Link
+        style={{ fontWeight: 'var(--pf-t--global--font--weight--body--bold)' }}
+        to={globalPipelineRunDetailsRoute(project, runId)}
+      >
+        {runName}
+      </Link>
+    ) : (
+      <b>{runName}</b>
+    );
+
+    const renderProject = registeredPipelineProject
+      ? registeredPipelineProject.metadata.annotations?.['openshift.io/display-name'] ??
+        registeredPipelineProject.metadata.name
+      : project;
+
+    return (
+      <Flex
```

Comment:
Based on my comment in ModelVersionDetailsView, let's wrap this returned Flex in `<DashboardDescriptionListGroup title="Registered from">`.